lib/Doctrine/ORM/EntityRepository.php

```diff
@@ -181,12 +181,13 @@ public function findAll()
     /**
      * Finds entities by a set of criteria.
      *
+     * @psalm-param array<string, mixed> $criteria
+     * @psalm-param list<string>|null $orderBy
      * @param int|null $limit
      * @param int|null $offset
      *
-     * @psalm-param array<string, mixed> $criteria
-     * @psalm-param list<string>|null $orderBy
-     * @psalm-return list<T> The objects.
+     * @return T[] The objects.
```

Comment:
Do IDEs understand generics?

Reply:
No, good catch. Didn't notice it came from a template annotation. I'll fix it.